servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3829,9 +3829,31 @@ public void deleteRole(ResourceContext ctx, String domainName, String roleName,
             throw ZMSUtils.requestError("deleteRole: admin role cannot be deleted", caller);
         }
 
+        /* we are not going to allow any user to delete
+         * role that related to any assertion of any policy */
+
+        Policies policies = getPolicies(ctx, domainName, true, true);
```

Comment:
getPolicies is an API endpoint so we can't call it within another api endpoint. This will screw up our counters as each deleteRole API will also count as getPolicies API as well. Take a look at the implementation of the getPolicies API and replace with api call with an appropriate internal method implementation.